lib/ansible/modules/packaging/language/maven_artifact.py

```diff
@@ -281,7 +281,13 @@ def download(self, artifact, destination):
             self.module.fail_json(msg='Copy failed. %s' % (':'.join([artifact.group_id, artifact.artifact_id, artifact.classifier, artifact.extension, artifact.url, destination, str(why)])))
 
     def checksum(self, artifact, dest):
-        return True # There is nothing to check if the artifact is local
+        if not os.path.isfile(dest):
+            return False
+        with open(create_artifact_location(artifact), 'rb') as file:
```

Comment:
Hi,
The method `create_artifact_location` doesn't actually yet. Sorry that I wasn't explicit in my comment (https://github.com/ansible/ansible/pull/19726#issuecomment-283656479).
I extracted the file path generation code from the `download` method into a new function:

```python
def create_artifact_location(artifact):
      return posixpath.join(self.repo, artifact.url).replace('file:/', '').replace('//', '/')
```